Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -24,6 +30,132 @@ namespace AzToolsFramework
+                            "LocalViewBookmarkLoader", false, "Trying to set an invalid index in a Vector3, index = %d", currentIndex);
+                        break;
+                    }
+                };
+
+                if (bookmarkType == LastKnownLocationKey)
+                {
+                    if (!m_lastKnownLocation.has_value())
+                    {
+                        m_lastKnownLocation.emplace(ViewBookmark{});
+                    }
+
+                    const int currentIndex = AZStd::stoi(AZStd::string(valueIndex));
+                    if (dataType == "Position")
+                    {
+                        setVec3Fn(m_lastKnownLocation->m_position, currentIndex);
+                    }
+                    else if (dataType == "Rotation")
+                    {
+                        setVec3Fn(m_lastKnownLocation->m_rotation, currentIndex);
+                    }
+                }
+                else if (bookmarkType == LocalBookmarksKey)
+                {
+                    if (auto existingBookmarkEntry = m_bookmarkMap.find(localBookmarksId.value());
+                        existingBookmarkEntry != m_bookmarkMap.end())
+                    {
+                        AZStd::vector<ViewBookmark>& bookmarks = existingBookmarkEntry->second;
+                        // if it is the first bookmark and it is the Position data it means it is the first one
+                        // and we have to create the Bookmark.
+                        if (valueIndex == "0" && dataType == "Position")
+                        {
+                            ViewBookmark bookmark;
+                            bookmark.m_position.SetX(aznumeric_cast<float>(value));
+                            bookmarks.push_back(bookmark);
+                        }
+                        else
+                        {
+                            const int bookmarkIndex = stoi(AZStd::string(bookmarkIndexStr->data()));
+                            AZ_Assert(bookmarkIndex < bookmarks.size(), "Bookmark index is out of bounds");
+                            ViewBookmark& bookmark = bookmarks[bookmarkIndex];
+                            const int currentIndex = stoi(AZStd::string(valueIndex));
+
+                            if (dataType == "Position")
+                            {
+                                setVec3Fn(bookmark.m_position, currentIndex);
+                            }
+                            else if (dataType == "Rotation")
+                            {
+                                setVec3Fn(bookmark.m_rotation, currentIndex);
+                            }
+                        }
+                    }
+                }
+            }
+        };
+
+        AZ::SettingsRegistryInterface::FixedValueString m_viewBookmarksKey;
+        AZStd::unordered_map<AZStd::string, AZStd::vector<ViewBookmark>> m_bookmarkMap;
+        AZStd::optional<ViewBookmark> m_lastKnownLocation;
+    };
```

Comment:
FYI: This function is the same as before and has just been moved out of line (before it was embedded in a function) to make the function easier to read at first glance